### PR TITLE
domzilla-caffeine 1.4.1 (new cask)

### PR DESCRIPTION
**Pull request by toobuntu**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by toobuntu** [comment]
This is a fork of cask `caffeine` with a universal binary that runs natively on Apple Silicon. Its last commit was [about three months ago](https://github.com/domzilla/Caffeine/commits/master/).

Cask `caffeine` provides only an x86_64 binary and requires Rosetta 2 to run on Apple Silicon. Its last commit was [about four years ago](https://github.com/IntelliScape/caffeine/commits/master/). It was itself a fork of the original, which ceased development.

From the homepage (excerpted):

> - **Is this the same Caffeine that I've used before?**
> Yes! Tomas Franzén of Lighthead Software originally developed Caffeine in 2006 ...
> In 2018, Michael Jones (IntelliScape Computer Solutions) reached out to Tomas to inquire if they could continue development of Caffeine.
> Tomas has graciously provided the source code under an open source license, allowing IntelliScape Computer Solutions to continue developing Caffeine where he left off.
> In 2022 we started modernizing the codebase and added support for macOS 11 and above. We also translated the application into several languages.

In https://github.com/Homebrew/homebrew-cask/pull/85502, the original `caffeine` cask description was replaced by the Intelliscape fork. The `intelliscape-caffeine` cask was deleted in https://github.com/Homebrew/homebrew-cask/pull/86270. Perhaps that should happen again with this fork, but I suppose that is a different discussion.

There seems to be no good livecheck for this. Maybe it should be skipped altogether.

File: Casks/d/domzilla-caffeine.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  on_sonoma :or_newer do
+    version "1.4.1"
+    sha256 "d96b375d0eb01f6cbce1f49e70fa484facf8be0236350f677249bacb7fe9cb87"
+
+    livecheck do
+      url :url
+      strategy :extract_plist do |items|
+        items["net.domzilla.caffeine"].short_version
+      end
+    end
```
Comment (bevanjkay) [inline Casks/d/domzilla-caffeine.rb:27]:
```suggestion
    livecheck do
      url "https://www.caffeine-app.net/download"
      strategy :header_match
    end
```

Comment (toobuntu) [reply Casks/d/domzilla-caffeine.rb]: Done! That's fantastic. Thank you.

**Comment by bevanjkay** [comment]
CI doesn't run for `sonoma` but I have confirmed that the audit passes locally.

**Review by bevanjkay — APPROVED** [review]

**Review by razvanazamfirei — APPROVED** [review]

**Pull request merged**